lib/spack/spack/util/environment.py

```diff
@@ -1075,33 +1075,42 @@ def environment_after_sourcing_files(
             only when the previous command succeeds (default: ``&&``)
     """
     # Set the shell executable that will be used to source files
-    shell_cmd = kwargs.get("shell", "/bin/bash")
-    shell_options = kwargs.get("shell_options", "-c")
-    source_command = kwargs.get("source_command", "source")
-    suppress_output = kwargs.get("suppress_output", "&> /dev/null")
+    if sys.platform == "win32":
+        shell_cmd = kwargs.get("shell", "cmd.exe")
+        shell_options = kwargs.get("shell_options", "/C")
+        suppress_output = kwargs.get("suppress_output", "")
+        source_command = kwargs.get("source_command", "")
+    else:
+        shell_cmd = kwargs.get("shell", "/bin/bash")
+        shell_options = kwargs.get("shell_options", "-c")
+        suppress_output = kwargs.get("suppress_output", "&> /dev/null")
+        source_command = kwargs.get("source_command", "source")
```

Comment:
If I remember correctly, there were two issues with this section of code. Firstly, we do not accept space separated arguments to Executable anymore. This function had to be reworked a bit to account for that. Secondly, it did not use a windows shell for windows.

I'm not sure about `%oneapi`. I don't know the context for that. I do know that I was running into similar problems when enabling environment tests and had to make similar code changes. I think this PR had better changes though. @johnwparent  for awareness since you'll be picking up [the environment PR](https://github.com/spack/spack/pull/35656)